comp/core/flare/helpers/perm_info_win.go

```diff
@@ -8,13 +8,261 @@
+		return nil, fmt.Errorf("unsupported AceType '%s': %x", fileName, ace.AceType)
+	}
+
+	return ace, nil
+}
+
+func sidToUserName(sid *windows.SID) string {
+	user, domain, _, err := sid.LookupAccount("")
+	if err == nil {
+		if len(domain) > 0 {
+			return fmt.Sprintf("%s\\%s", domain, user)
+		}
+
+		return user
+	}
+
+	return sid.String()
+}
+
+// This function attempts to generate the same rights abbreviation as built-in icacls.exe (when
+// icacls.exe runs without arguments it will generate abbreviations legend). Compatibility is
+// not complete, e.g. no attempts to interpret directory rights, however numeric values of mask
+// are reported. More details can be found here
+//     https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/icacls
+func accessMaskToStr(m uint32) string {
+	if (m & DD_FILE_FULL) == DD_FILE_FULL {
+		return "(F)"
+	}
+	if (m & DD_FILE_MODIFY) == DD_FILE_MODIFY {
+		return "(M)"
+	}
+	if (m & DD_FILE_READ_EXEC_WRITE) == DD_FILE_READ_EXEC_WRITE {
+		return "(RX,W)"
+	}
+	if (m & DD_FILE_READ_EXEC) == DD_FILE_READ_EXEC {
+		return "(RX)"
+	}
+	if (m & DD_FILE_WRITE) == DD_FILE_WRITE {
+		return "(W)"
+	}
+	if m == windows.GENERIC_ALL {
+		return "(F)"
+	}
+	if m&(windows.GENERIC_READ|windows.GENERIC_WRITE|windows.GENERIC_EXECUTE) != 0 {
+		rights := make([]string, 0, 3)
+		if (m & windows.GENERIC_READ) == windows.GENERIC_READ {
+			rights = append(rights, "GR")
+		}
+		if (m & windows.GENERIC_WRITE) == windows.GENERIC_WRITE {
+			rights = append(rights, "GW")
+		}
+		if (m & windows.GENERIC_EXECUTE) == windows.GENERIC_EXECUTE {
+			rights = append(rights, "GE")
+		}
+		return fmt.Sprintf("(%s)", strings.Join(rights, ","))
+	}
+
+	return ""
+}
 
-func (p permissionsInfos) add(filePath string) error {
-	return nil
+// This function attempts to generate the same rights abbreviation as built-in icacls.exe (when
+// icacls.exe runs without arguments it will generate abbreviations legend). Numeric values of flags
+// are reported. More details can be found here
+//     https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/icacls
+func accessFlagToStr(f uint8) string {
+	if (f & windows.OBJECT_INHERIT_ACE) == windows.OBJECT_INHERIT_ACE {
+		return "(OI)"
+	}
+	if (f & windows.CONTAINER_INHERIT_ACE) == windows.CONTAINER_INHERIT_ACE {
+		return "(CI)"
+	}
+	if (f & windows.NO_PROPAGATE_INHERIT_ACE) == windows.NO_PROPAGATE_INHERIT_ACE {
+		return "(NP)"
+	}
+	if (f & windows.INHERIT_ONLY_ACE) == windows.INHERIT_ONLY_ACE {
+		return "(IO)"
+	}
+	if (f & windows.INHERITED_ACE) == windows.INHERITED_ACE {
+		return "(I)"
+	}
+
+	return ""
+}
+
+func denyToStr(aceType uint8) string {
+	if aceType == winutil.ACCESS_DENIED_ACE_TYPE {
+		return "(DENY)"
+	}
+
+	return ""
+}
+
+func (p permissionsInfos) addAgetExeFiles() {
+	// Get Datadog bin directory (optional if err)
+	installDir, err := winutil.GetProgramFilesDirForProduct("DataDog Agent")
```

Comment:
It is not since it is using registry API call. Also I have copied this line from our existing and used flare code https://github.com/DataDog/datadog-agent/blob/main/pkg/flare/archive_win.go#L206. Plus tested it.

Reply:
would it be better to use the existing wrapper/helper
https://github.com/DataDog/datadog-agent/blob/main/pkg/util/winutil/shutil.go#L29-L31